webboot/webboot.go

```diff
@@ -26,43 +33,134 @@ var (
 	ipv4      = flag.Bool("ipv4", true, "use IPV4")
 	ipv6      = flag.Bool("ipv6", true, "use IPV6")
 	osSystems = map[string]*webboot.Distro{
-		"tinycore": &webboot.Distro{"/boot/vmlinuz64", "/boot/corepure64.gz", "console=tty1", "https:louis.com"},
-		"TestISO":  &webboot.Distro{"/kernel", "/initiso.cpi", "console=ttyS0 console=tty0", "https:louis.com"},
+		"tinycore": &webboot.Distro{"/boot/vmlinuz64", "/boot/corepure64.gz", "console=tty1", "http://tinycorelinux.net/10.x/x86_64/release/CorePure64-10.1.iso"},
+		"TestISO":  &webboot.Distro{"/boot/kernel", "/boot/initiso.cpi", "console=ttyS0 console=tty0", "http://releases.ubuntu.com/18.04.2/ubuntu-18.04.2-desktop-amd64.iso"},
 	}
 )
 
+// parseArgs takes the arg user passes e.g. name of OS or link
+// if arg matches the name of one of the supported OSses, parseArgs retrieves its link from the map & assigns filename to be arg.
+// The steady filename allows it to be consistent with the key in the supported OSses map as well as the one the user described the OS as
+// if the user passes a valid URL, parseArgs calls name to extract the filename from a parsed link and returns a valid link with the filename
+func parseArgs(arg string) (string, string, error) {
+	var link, filename string
+	if u, ok := osSystems[arg]; ok {
+		link = u.DownloadLink
+		filename = arg
+	} else {
+		var err error
+		filename, err = name(arg)
+		link = arg
+		if err != nil {
+			return "", "", fmt.Errorf("Invalid URL: got %v, want a valid URL. Got %v error, want nil", arg, err)
+		}
+	}
+	return link, filename, nil
+}
+
+// linkOpen returns an io.ReadCloser that holds the content of the OS
+func linkOpen(link string) (io.ReadCloser, error) {
+	log.Println("Retrieving the file")
+	resp, err := http.Get(link)
+	if err != nil {
+		return nil, err
+	}
+	// checks if request to receive URL is successful
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("non-200 HTTP status: %d", resp.StatusCode)
+	}
+	return resp.Body, nil
+}
+
+// write copies the content from the downloaded ISO into a new file with filename.
+func write(read io.ReadCloser, filename string) error {
```

Comment:
that is true. It could be any downloaded file that the user asked for at this point. However, later on, it will be caught and the user will receive an error if the file the user asked is not an OS, it will not work

Reply:
I changed the comment :) this does provide a great description